samples/api/hello_triangle_1_3/hello_triangle_1_3.cpp

```diff
@@ -0,0 +1,1313 @@
+	}
+	return VK_FALSE;
+}
+#endif
+
+/**
+ * @brief Validates a list of required extensions, comparing it with the available ones.
+ *
+ * @param required A vector containing required extension names.
+ * @param available A VkExtensionProperties object containing available extensions.
+ * @return true if all required extensions are available
+ * @return false otherwise
+ */
+bool HelloTriangleV13::validate_extensions(const std::vector<const char *>          &required,
+                                           const std::vector<VkExtensionProperties> &available)
+{
+	bool all_found = true;
+
+	for (const auto *extension_name : required)
+	{
+		bool found = false;
+		for (const auto &available_extension : available)
+		{
+			if (strcmp(available_extension.extensionName, extension_name) == 0)
+			{
+				found = true;
+				break;
+			}
+		}
+
+		if (!found)
+		{
+			// Output an error message for the missing extension
+			LOGE("Error: Required extension not found: {}", extension_name);
+			all_found = false;
+		}
+	}
+
+	return all_found;
+}
+
+/**
+ * @brief Validates a list of required layers, comparing it with the available ones.
+ *
+ * @param required A vector containing required layer names.
+ * @param available A VkLayerProperties object containing available layers.
+ * @return true if all required extensions are available
+ * @return false otherwise
+ */
+bool HelloTriangleV13::validate_layers(const std::vector<const char *>      &required,
+                                       const std::vector<VkLayerProperties> &available)
+{
+	bool all_found = true;
+
+	for (const auto *layer_name : required)
+	{
+		bool found = false;
+		for (const auto &available_layer : available)
+		{
+			if (strcmp(available_layer.layerName, layer_name) == 0)
+			{
+				found = true;
+				break;
+			}
+		}
+
+		if (!found)
+		{
+			LOGE("Error: Required layer not found: {}", layer_name);
+			all_found = false;
+		}
+	}
+
+	return all_found;
+}
+
+/**
+ * @brief Initializes the Vulkan instance.
+ */
+void HelloTriangleV13::init_instance()
+{
+	LOGI("Initializing Vulkan instance.");
+
+	if (volkInitialize())
+	{
+		throw std::runtime_error("Failed to initialize volk.");
+	}
+
+	uint32_t instance_extension_count;
+	VK_CHECK(vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, nullptr));
+
+	std::vector<VkExtensionProperties> available_instance_extensions(instance_extension_count);
+	VK_CHECK(vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, available_instance_extensions.data()));
+
+	std::vector<const char *> required_instance_extensions{VK_KHR_SURFACE_EXTENSION_NAME};
+
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	bool has_debug_utils = false;
+	for (const auto &ext : available_instance_extensions)
+	{
+		if (strncmp(ext.extensionName, VK_EXT_DEBUG_UTILS_EXTENSION_NAME, strlen(VK_EXT_DEBUG_UTILS_EXTENSION_NAME)) == 0)
+		{
+			has_debug_utils = true;
+			break;
+		}
+	}
```

Comment:
Similar to the above:
```
	bool has_debug_utils = (std::any_of(
	    available_instance_extensions.begin(),
	    available_instance_extensions.end(),
	    [](auto const &ext) { return strncmp(ext.extensionName, VK_EXT_DEBUG_UTILS_EXTENSION_NAME, strlen(VK_EXT_DEBUG_UTILS_EXTENSION_NAME)) == 0; }));
```

Reply:
Same here. While that may be modern C++, it's imo harder to read.

Reply:
Well, I think an explicit loop is harder to read than an algorithm.